x-pack/plugins/apm/public/components/app/mobile/charts/mobile_crashes_treemap/index.tsx

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState } from 'react';
+import { EuiSpacer } from '@elastic/eui';
+import { TreemapSelect, TreemapTypes } from './treemap_select';
+import { TreemapChart } from '../../../../shared/charts/treemap_chart';
+import { useFetcher } from '../../../../../hooks/use_fetcher';
+import {
+  DEVICE_MODEL_IDENTIFIER,
+  SERVICE_VERSION,
+} from '../../../../../../common/es_fields/apm';
+
+const ES_FIELD_MAPPING: Record<TreemapTypes, string> = {
+  [TreemapTypes.Devices]: DEVICE_MODEL_IDENTIFIER,
+  [TreemapTypes.Versions]: SERVICE_VERSION,
+};
+
+export function MobileCrashesTreemap({
+  kuery,
+  serviceName,
+  start,
+  end,
+  environment,
+}: {
+  kuery: string;
+  serviceName: string;
+  start: string;
+  end: string;
+  environment: string;
+}) {
+  const [selectedTreemap, selectTreemap] = useState(TreemapTypes.Devices);
+
+  const { data, status } = useFetcher(
+    (callApmApi) => {
+      const fieldName = ES_FIELD_MAPPING[selectedTreemap];
+      if (fieldName) {
+        return callApmApi(
+          'GET /internal/apm/mobile-services/{serviceName}/error_terms',
+          {
+            params: {
+              path: {
+                serviceName,
+              },
+              query: {
+                environment,
+                kuery: `${kuery} and error.type: crash`,
```

Comment:
The only difference between `MobileCrashesTreemap` and `MobileErrorTreemap` is the kuery.

Can you also create a shared component for both and pass the kuery when you're caliin the component?. As you already do here: https://github.com/elastic/kibana/pull/165892/files#diff-63528f584bbeb33f1c81d92709a7d502604faaf9665147b1f0e67880aacb5e8fR250

In the UI, we try to create abstract components that can be shared and reused across the app.